concourse/scripts/fly_sync_and_login.sh

```diff
@@ -2,19 +2,47 @@
 set -euo pipefail
 
 # Required env vars
-# CONCOURSE_URL
-# CONCOURSE_ATC_USER
-# CONCOURSE_ATC_PASSWORD
-# FLY_CMD
-# FLY_TARGET
+# shellcheck disable=SC2086
+: $CONCOURSE_URL \
+  $CONCOURSE_ATC_USER \
+  $CONCOURSE_ATC_PASSWORD \
+  $FLY_CMD \
+  $FLY_TARGET
 
-FLY_CMD_URL="${CONCOURSE_URL}/api/v1/cli?arch=amd64&platform=$(uname | tr '[:upper:]' '[:lower:]')"
-echo "Downloading fly command..."
-curl "$FLY_CMD_URL" -# -L -f -z "$FLY_CMD" -o "$FLY_CMD" -u "${CONCOURSE_ATC_USER}:${CONCOURSE_ATC_PASSWORD}"
-chmod +x "$FLY_CMD"
+fetch_fly() {
+  echo "Downloading fly .."
+  FLY_CMD_URL="${CONCOURSE_URL}/api/v1/cli?arch=amd64&platform=$(uname | tr '[:upper:]' '[:lower:]')"
+  curl \
+    --progress-bar \
+    --location \
+    --fail \
+    --output "$FLY_CMD" \
+    "$FLY_CMD_URL"
+  chmod +x "$FLY_CMD"
+}
 
-echo "Doing fly login"
-$FLY_CMD -t "${FLY_TARGET}" login --concourse-url "${CONCOURSE_URL}" -u "${CONCOURSE_ATC_USER}" -p "${CONCOURSE_ATC_PASSWORD}"
+fly_sync() {
+  echo "Doing fly sync .."
+  $FLY_CMD -t "${FLY_TARGET}" sync
+}
 
-echo "Doing fly sync"
-$FLY_CMD -t "${FLY_TARGET}" sync
+fly_login() {
+  echo "Doing fly login .."
+  $FLY_CMD -t "${FLY_TARGET}" login --concourse-url "${CONCOURSE_URL}" -u "${CONCOURSE_ATC_USER}" -p "${CONCOURSE_ATC_PASSWORD}"
+}
+
+fly_is_runnable() {
+  [ -x "${FLY_CMD}" ]
+}
+
+if ! fly_is_runnable; then
```

Comment:
This construct is effectively an `unless...else...` block. Can we reverse this so that it's easier to read/follow.

Reply:
I don't agree that `unless .. else..` is more readable than `if !condition .. else`. But I don't feel strongly about it :-) Your call.